src/main/java/nl/ordina/jobcrawler/service/VacancyService.java

```diff
@@ -23,65 +27,82 @@ public VacancyService(VacancyRepository vacancyRepository, SkillService skillSer
     }
 
 
-    //******** Adding ********//
-    // Adding vacancy and accompanying skills and relations to the database //
-    public Vacancy add(Vacancy vacancy) {
-        /**
-         * The function below which is commented out, is used to prevent duplicate skills in the skills table.
-         * For some reason it throws a hibernate exception; detached entity passed to persist
-         * javax.persistence.PersistenceException
-         * We can solve this using an EntityManager. Implementing this will take some time while after the mvp we might put the entire vacancy body into a single column. By commenting out the lines below we allow duplicates.
-         * The unique parameter in Skill.java is set to false. Someone using the application won't notice the difference.
-         */
-        
-        //replacing the skills with existing skills from the database if present
-//        Set<Skill> existingSkills = skillService.linkToExistingSkills(vacancy.getSkills());
-//        existingSkills.forEach(skill -> skill.addVacancy(vacancy)); // add the vacancy to the skill
-//        vacancy.setSkills(existingSkills); // add the skills to the vacancy
-
-
-        if (vacancy.hasValidURL()){    //checking the url, if it is malformed it will throw a VacancyURLMalformedException
-            return vacancyRepository.saveAndFlush(vacancy);
-        } else {
-            throw new VacancyURLMalformedException("Website could not be reached");
-        }
-
-
+    /**
+     * Returns the vacancy with the specified id.
+     *
+     * @param id ID of the vacancy to retrieve.
+     * @return An optional of the requested vacancy if found, and an empty optional otherwise.
+     */
+    @Override
+    public Optional<Vacancy> findById(UUID id) {
+        return vacancyRepository.findById(id);
     }
 
-    //******** Getting ********//
-    public List<Vacancy> getAllVacancies() {
+
+    /**
+     * Returns all vacancies in the database.
+     *
+     * @return All vacancies in the database.
+     */
+    @Override
+    public List<Vacancy> findAll() {
         return vacancyRepository.findAll();
     }
 
-    public Optional<Vacancy> getVacancyByID(UUID id) {
-        return vacancyRepository.findById(id);
-    }
 
-    public Set<Vacancy> getVacanciesBySkill(String skill) {
-        return skillService.getVacanciesBySkill(skill);
+    /**
+     * Updates the specified vacancy, identified by its id.
+     *
+     * @param vacancy The vacancy to update.
+     * @return True if the update succeeded, otherwise false.
+     */
+    @Override
+    public boolean update(Vacancy vacancy) {
```

Comment:
Dit lijkt me makkelijk en snel te implementeren. Neem dit nog aub mee in deze branch, voordat je em afrond.